src/app/components/recipes/recipes.effects.ts

```diff
@@ -0,0 +1,52 @@
+import { Injectable } from '@angular/core';
+import { Actions, createEffect, ofType } from '@ngrx/effects';
+import { tap, map, concatMap } from 'rxjs/operators';
+import { Router } from '@angular/router';
+import {
+  loadAllRecipes,
+  allRecipesLoaded,
+  editRecipe,
+  deleteRecipe,
+} from './recipes.actions';
+import { ApiService } from '../api.service';
+
+@Injectable()
+export class RecipesEffects {
+  loadRecipes$ = createEffect(() =>
+    this.actions$.pipe(
+      ofType(loadAllRecipes),
```

Comment:
As above, we should check if it's already in the store via withLatestFrom & filter